test/testdrive/dates-times.td

```diff
@@ -35,27 +42,68 @@ $ set schema={
 $ kafka-create-topic topic=data
 
 $ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
-{"d": 18135, "ts": 1566865029000000}
-{"d": 0, "ts": 65000000}
+{"d": 18135, "ts_millis": 1566865029000000, "ts_micros": 1566865029000000}
+{"d": 0, "ts_millis": 65000000, "ts_micros": 65000000}
 
 > CREATE CONNECTION kafka_conn
   TO KAFKA (BROKER '${testdrive.kafka-addr}');
 
+> CREATE CONNECTION csr_conn TO CONFLUENT SCHEMA REGISTRY (
+    URL '${testdrive.schema-registry-url}'
+  );
+
 > CREATE SOURCE data FROM
   KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-data-${testdrive.seed}')
   FORMAT AVRO USING SCHEMA '${schema}'
 
 > CREATE MATERIALIZED VIEW data_view as SELECT * from data
 
 > SELECT * FROM data_view
-2019-08-27 "2019-08-27 00:17:09"
-1970-01-01 "1970-01-01 00:01:05"
-
-# Since the latest record in the topic is 10, the logical timestamp at which
-# we read should be 10 - 1.
-# This is no longer true after timestamping - commenting out
-#> SELECT mz_now() FROM data_view LIMIT 1
-#9
+1970-01-01 "1970-01-01 18:03:20" "1970-01-01 00:01:05"
+2019-08-27 "+51621-12-22 21:50:00" "2019-08-27 00:17:09"
+
+> CREATE SINK data_sink FROM data
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-data-roundtrip-${testdrive.seed}')
+  KEY (d) NOT ENFORCED
+  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+  ENVELOPE UPSERT
+
+$ schema-registry-verify schema-type=avro subject=testdrive-data-roundtrip-${testdrive.seed}-value
+{"type":"record","name":"envelope","fields":[{"name":"d","type":{"type":"int","logicalType":"date"}},{"name":"ts_millis","type":{"type":"long","logicalType":"timestamp-millis"}},{"name":"ts_micros","type":{"type":"long","logicalType":"timestamp-micros"}}]}
+
+$ kafka-verify-data format=avro sink=materialize.public.data_sink sort-messages=true
+{"d": 0} {"d": 0, "ts_millis": 65000000, "ts_micros": 65000000}
+{"d": 18135} {"d": 18135, "ts_millis": 1566865029000000, "ts_micros": 1566865029000000}
+
+> CREATE MATERIALIZED VIEW ts_precision AS
+  SELECT
+    TRUE AS key,
+    '1970-01-01T00:00:00.123456'::timestamp AS ts,
+    '1970-01-01T00:00:00.123456'::timestamp(1) AS ts1,
+    '1970-01-01T00:00:00.123456'::timestamp(2) AS ts2,
+    '1970-01-01T00:00:00.123456'::timestamp(3) AS ts3,
+    '1970-01-01T00:00:00.123456'::timestamp(4) AS ts4,
+    '1970-01-01T00:00:00.123456'::timestamp(5) AS ts5,
+    '1970-01-01T00:00:00.123456'::timestamp(6) AS ts6,
```

Comment:
should at least one value > 6

Reply:
That would result in an error https://github.com/MaterializeInc/materialize/pull/21570/files#diff-5109487703f4f16f3f789284c05a46f64b76a469678cfd611b2b78a92e65d3ebR137.

Adding it as a separate select statement.

Reply:
Added!

Reply:
Oh, interesting that PostgreSQL just downgrades to the max precision. I like our behavior of erroring better. We can always adjust to match the PostgreSQL behavior in the future, if it becomes an issue.

Reply:
Sounds good. Makes sense to go from strict to relaxed instead of the other way round.